tensorflow_addons/layers/tests/spectral_normalization_test.py

```diff
@@ -0,0 +1,173 @@
+# Copyright 2020 The TensorFlow Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# =============================================================================
+
+import tempfile
+from pathlib import Path
+
+import numpy as np
+import pytest
+import tensorflow as tf
+
+from tensorflow_addons.layers import spectral_normalization
+from tensorflow_addons.utils import test_utils
+
+
+def test_keras():
+    input_data = np.random.random((10, 3, 4)).astype(np.float32)
+    test_utils.layer_test(
+        spectral_normalization.SpectralNormalization,
+        kwargs={"layer": tf.keras.layers.Dense(2), "input_shape": (3, 4)},
+        input_data=input_data,
+    )
+
+
+def test_from_to_config():
+    base_layer = tf.keras.layers.Dense(1)
+    sn = spectral_normalization.SpectralNormalization(base_layer)
+    config = sn.get_config()
+
+    new_sn = spectral_normalization.SpectralNormalization.from_config(config)
+    assert sn.power_iterations == new_sn.power_iterations
+
+
+def test_save_load_model():
+    base_layer = tf.keras.layers.Dense(1)
+    input_shape = [1]
+
+    inputs = tf.keras.layers.Input(shape=input_shape)
+    sn_layer = spectral_normalization.SpectralNormalization(base_layer)
+    model = tf.keras.models.Sequential(layers=[inputs, sn_layer])
+
+    # initialize model
+    model.predict(np.random.uniform(size=(2, 1)))
+
+    with tempfile.TemporaryDirectory() as model_dir:
+        model_path = str(Path(model_dir) / "saved_model")
+        model.save(model_path)
+        new_model = tf.keras.models.load_model(model_path)
```

Comment:
One minor thing. We can use pytest fixture here. We can change the function signature to
```python
def test_save_load_model(tmpdir):
```
The example usage can be found here
https://github.com/tensorflow/addons/blob/master/tensorflow_addons/losses/tests/kappa_loss_test.py#L95-L103

Reply:
Nice, I didn't know about this